protocol/interop-access-list.md

```diff
@@ -0,0 +1,103 @@
+
+### Static Declaration
+
+Ethereum transactions can use access lists to specify storage slots accessed during execution, helping clients prefetch data and reduce runtime costs. While rarely used today, cross-chain message relayers will need to adopt new patterns. However, this only affects the execution of a cross-chain transaction, not the initiation, and most users will rely on relayers with the necessary tooling.
+
+Static declaration also enables on-chain reverts per message, requiring updates to the Supervisor and `CrossL2Inbox` (detailed in [the specs](https://github.com/ethereum-optimism/specs/pull/612)). Access lists will warm a checksum slot tied to an `Identifier` and `msgHash` commitment. If warm, the commitment is valid; otherwise, the transaction reverts.
+
+The access list will also include other types of `storageKeys` that warm slots on the `CrossL2Inbox` contract. While these slots won’t be used by the implementation, they are important for security considerations. These types, detailed in the specs, are necessary for the client to perform correct lookups. Our primary focus is the checksum key, as it’s the one we check for warmth on-chain.
+
+### Gas introspection
+
+Gas introspection lets contracts check if a storage slot was warmed by measuring its access cost. Warmed slots are cheaper to read, so contracts can verify if a cross-chain message was properly declared in the access list. If not, the `validateMessage` call reverts. This ensures efficient enforcement of access list rules with minimal overhead.
+
+The main downside with this approach is that it couples key security properties to the resource schedule, which is subject to change over time. Keeping the cognitive overhead of having to test a codepath every time that the gas schedule changes can create pain.
+
+### Implementation example
+
+The `WARM_READ_THRESHOLD` value is up for debate, the cost difference between hot and cold is very large, but it would be good to have a consensus over it. The `calculateChecksum` will be expanded on the specs doc.
+
+```solidity
+uint256 internal constant WARM_READ_THRESHOLD = 150;
+
+function _isWarm(bytes32 _slot) internal view returns (bool isWarm, uint256 result) {
+    assembly {
+        let startGas := gas()
+        result := sload(_slot)
+        let endGas := gas()
+        isWarm := iszero(gt(sub(startGas, endGas), WARM_READ_COST))
+    }
+ }
+
+function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
+    bytes32 checksum = calculateChecksum(_id, _msgHash);
+
+    (bool _isSlotWarm,) = _isWarm(checksum);
+
+    if (!_isSlotWarm) revert NotWarm();
```

Comment:
Important detail from previous discussion to document here: the storage-warmup is rolled back when the `revert` happens. That means that prior invalid message lookups will not cause later lookups of the same checksum to appear valid.